src/main/java/de/uniba/dsg/serverless/azure/AzureLogHandler.java

```diff
@@ -83,80 +79,132 @@ public void writePerformanceDataToFile(String fileName) {
+							startupDuration, duration, -1, -1, -1);
+					performanceData.add(data);
+				}
 			}
-			performanceDataList.addAll(getPerformanceData(fileContent));
+
+		} catch (IOException e) {
+			throw new SeMoDeException("Exception while parsing requests via REST API from Application Insights", e);
 		}
-		return performanceDataList;
+
+		return performanceData;
 	}
 
-	private List<PerformanceData> getPerformanceData(String fileContent) {
-		// Split the text into lines
-		String lines[] = fileContent.split("\\r?\\n");
-
-		// Stores the start times of functions that are not completed
-		Map<String, LocalDateTime> pendingData = new HashMap<>();
-
-		List<PerformanceData> result = new ArrayList<>();
-
-		for (String line : lines) {
-			String[] lineParts = line.split(" ", 3);
-			String message = lineParts[2];
-			if (message.startsWith(FUNCTION_STARTED)) {
-				// Start message
-				String id = AzureLogAnalyzer.extractRequestId(message);
-				LocalDateTime startTime = AzureLogAnalyzer.parseTime(lineParts[0]);
-				pendingData.put(id, startTime);
-			} else if (message.startsWith(FUNCTION_COMPLETED)) {
-				// Finish Message
-				String id = AzureLogAnalyzer.extractRequestId(message);
-				LocalDateTime startTime = pendingData.remove(id);
-				double duration = AzureLogAnalyzer.extractDuration(message);
-				PerformanceData data = new PerformanceData(functionName, "", id, startTime, duration, -1, -1, -1);
-				result.add(data);
+	private Map<String, Double> getHostStartupDurations() throws SeMoDeException {
+		Map<String, Double> hostStartupDurations = new HashMap<>();
+
+		String functionTraces = getTracesAsJSON();
+
+		JsonFactory factory = new JsonFactory();
+		ObjectMapper mapper = new ObjectMapper(factory);
+
+		try {
+			JsonNode tableNode = mapper.readTree(functionTraces).get("tables").get(0);
+			JsonNode columnsNode = tableNode.get("columns");
+			JsonNode rowsNode = tableNode.get("rows");
+
+			Map<String, Integer> columnsIndex = parseColumns(columnsNode);
+
+			for (JsonNode rowNode : rowsNode) {
+				String message = rowNode.get(columnsIndex.get("message")).asText();
+
+				if (message.startsWith(MESSAGE_HOST_STARTED)) {
+					String container = rowNode.get(columnsIndex.get("cloud_RoleInstance")).asText();
+					double hostStartupDuration = AzureLogAnalyzer.parseHostStartupDuration(message);
+
+					hostStartupDurations.put(container, hostStartupDuration);
+				}
 			}
+		} catch (IOException e) {
+			throw new SeMoDeException("Exception while parsing traces via REST API from Application Insights", e);
 		}
-		return result;
+
+		return hostStartupDurations;
+	}
+
+	private String getRequestsAsJSON() throws SeMoDeException {
+		return runQuery("requests | order by timestamp asc");
 	}
 
-	private List<CloudFile> getLogFiles() throws SeMoDeException {
+	private String getTracesAsJSON() throws SeMoDeException {
+		return runQuery("traces | order by timestamp asc");
```

Comment:
traces 
| where message startswith "Host started" and timestamp > todatetime('2018-05-15T12:43:23.297') and timestamp < todatetime('2018-05-16T12:43:23.297')
| order by timestamp asc 

Reduces the logic within the source code. Filtering is done by the provider's API and adds a possibility to specify the interval, where the data is fetched.